master/docs/examples/gitlab.cfg

```diff
@@ -0,0 +1,223 @@
+# -*- python -*-
+# ex: set filetype=python:
+
+# This is a sample buildmaster config file. It must be installed as
+# 'master.cfg' in your buildmaster's base directory.
+
+from buildbot.plugins import *
+import re
+
+
+# Are all your projects built the same way?
+# Do you yearn for a way to do simple static configuration?
+# If so, try writing a function.
+# Here's an example that
+# - uses GitLab
+# - provides one regular builder and one smoke test builder per project
+# - assumes the projects use Gnu automake etc.
+#
+# To use this example with your own local instance of gitlab:
+#
+# 0. Edit this file to replace example.com with your own domain.
+#
+# 1. Set up local mirrors of the gnu projects, e.g.
+#   for proj in hello time
+#   do 
+#     git clone --mirror git@gitlab.com:GNU/$proj.git
+#     cd $proj
+#     git push --mirror git@gitlab.example.com:build/gnu-$proj.git
+#     cd ..
+#   done
+#
+# 2. Follow instructions below marked "CONFIGME: GitLab" to
+# configure GitLab to know about this buildbot.
+#
+# Your buildbot should then do automated builds of merge requests
+# that target forks of those local projects.
+
+def addGitLabProject(repourl, workernames, branch=None):
+    '''
+    Add a builder and a merge request builder for the given branch
+    of the given project, running on the given workers.
+    Give each a Force button.
+    Gets project name from the repourl, ignoring namespace.
+    '''
+
+    if branch is None:
+        branch = "master"
+    # Strip off everything before project name
+    # FIXME: parse this more artfully to allow projects in folders
+    name = re.sub(r'^.*/', '', repourl)
+    # Strip off .git suffix, if present
+    name = re.sub(r'\.git$', '', name)
+
+    # Regular builder
+    # Adjust this factory to match your site's build steps.
+    # This example uses the canonical gnu configure/make steps.
+    factory = util.BuildFactory()
+    factory.addStep(steps.GitLab(repourl=repourl, mode='incremental'))
+    factory.addStep(steps.ShellCommand(haltOnFailure=True, command=["sh", "-c", "if test -x ./bootstrap; then ./bootstrap; fi"]))
```

Comment:
haltOnFailure=True is the default.

Reply:
haltOnFailure is the default for the subclasses Configure and Compile, but the parent ShellCommand only defaults to flunkOnFailure = True.

Bootstrap failure is serious, so haltOnFailure=True is appropriate.

If you think ShellCommand's default should be haltOnFailure=True, please file a bug :-)

Reply:
I guess that is the kind of API mistake that is hard to fix now. We can't change default values like this.